### PR TITLE
Updated micronaut version, see https://github.com/eventuate-foundatio…

### PR DESCRIPTION
**Pull request by dartartem**
…n/eventuate-common/issues/81 #28: Upserts to mongodb some time fail. Fixed, added test.

### CODE REVIEW

File: order-history-service/src/main/java/io/eventuate/examples/tram/ordersandcustomers/orderhistoryservice/domain/CustomerViewRepository.java [hunk 1]
```diff
@@ -1,41 +1,85 @@
...
 
   public void addOrder(Long customerId, Long orderId, Money orderTotal) {
-    mongoTemplate.upsert(new Query(where("id").is(customerId)),
-            new Update().set("orders." + orderId, new OrderInfo(orderId, orderTotal)), CustomerView.class);
+    repeatOnFailure(() -> {
+      BasicDBObject orderInfo = new BasicDBObject()
+              .append("orderId", orderId)
+              .append("orderTotal", orderTotal.getAmount());
+
+      collection().findOneAndUpdate(new BasicDBObject("_id", customerId),
```
Comment (cer) [inline order-history-service/src/main/java/io/eventuate/examples/tram/ordersandcustomers/orderhistoryservice/domain/CustomerViewRepository.java]: This pattern is repeated throughout the code:

```
      collection().findOneAndUpdate(new BasicDBObject("_id", ID),
              new BasicDBObject("$set", NEW_VALUE),
              upsertOptions());
```

Make use of `AbstractRepository` and define a `findOneAndUpdate(ID, NEW_VALUE)` method that implements this pattern

File: order-history-service/src/main/java/io/eventuate/examples/tram/ordersandcustomers/orderhistoryservice/domain/CustomerViewRepository.java [hunk 2]
```diff
@@ -1,41 +1,85 @@
...
-
-  public CustomerViewRepository(MongoTemplate mongoTemplate) {
-    this.mongoTemplate = mongoTemplate;
+  public CustomerViewRepository(MongoClient mongoClient) {
+    super(mongoClient, "customers");
   }
 
   public Optional<CustomerView> findById(Long customerId) {
-    return Optional.ofNullable(mongoTemplate.findById(customerId, CustomerView.class));
+    Document customerDocument = collection()
```
Comment (cer) [inline order-history-service/src/main/java/io/eventuate/examples/tram/ordersandcustomers/orderhistoryservice/domain/CustomerViewRepository.java]: This pattern is repeated

```
    Document customerDocument = collection()
            .find(new BasicDBObject("_id", ID))
            .first();

    return Optional
            .ofNullable(customerDocument)

```

Make use of `AbstractRepository` and define a `findOne(ID)` method, which returns `Optional<Document>`

Comment (dartartem) [reply order-history-service/src/main/java/io/eventuate/examples/tram/ordersandcustomers/orderhistoryservice/domain/CustomerViewRepository.java]: ok

File: order-history-service/src/main/java/io/eventuate/examples/tram/ordersandcustomers/orderhistoryservice/domain/CustomerViewRepository.java (hunk 1 above)
Comment (dartartem) [reply order-history-service/src/main/java/io/eventuate/examples/tram/ordersandcustomers/orderhistoryservice/domain/CustomerViewRepository.java]: ok

**Pull request merged**